ConcurrentAPI.adoc

```diff
@@ -0,0 +1,220 @@
+== Distributed Concurrent API
+
+=== Motivation
+
+Coming from an initial issue where we need to provide an implementation for vert.x cluster manager for the shared data feature, provide concurrent and distributed programming capabilities.
+
+* Distributed locks -> is it a good idea ?
+* Distributed semaphores : first use case, vert-x cluster management
+* Distributed Countdown latch : it seems the team has already spoken about having it
+* Distributed Atomic Variables : we already have them, would be worth to offer an specific AtomicLong (for example) ?
```

Comment:
We have some AtomicObjectFactory, I don't know the details yet and apparently is a bit experimental